src/components/NcListItem/NcListItem.vue

```diff
@@ -320,28 +331,30 @@
 		:exact="to ? exact : null">
 		<li class="list-item__wrapper"
 			:class="{ 'list-item__wrapper--active' : isActive || active }">
-			<a :id="anchorId"
-				ref="list-item"
-				:href="routerLinkHref || href"
-				:target="target || (href === '#' ? undefined : '_blank')"
-				:rel="href === '#' ? undefined : 'noopener noreferrer'"
+			<div ref="list-item"
 				class="list-item"
-				:aria-label="linkAriaLabel"
+				tabindex="0"
 				@mouseover="handleMouseover"
 				@mouseleave="handleMouseleave"
 				@focus="handleFocus"
 				@blur="handleBlur"
 				@keydown.tab.exact="handleTab"
 				@click="onClick($event, navigate, routerLinkHref)"
 				@keydown.esc="hideActions">
-
-				<div class="list-item-content__wrapper"
-					:class="{ 'list-item-content__wrapper--compact': compact }">
+				<div :class="['list-item-content__wrapper', {
+					'list-item-content__wrapper--compact': compact
+				}]">
 					<!-- @slot This slot is used for the NcAvatar or icon -->
 					<slot name="icon" />
 
 					<!-- Main content -->
-					<div class="list-item-content">
+					<a :id="anchorId || undefined"
+						tabindex="-1"
```

Comment:
No longer possible to activate the link?

Reply:
To keep current behavior of focusing the wrapper (`tabindex="0"`) but not introduce a second element that can be focused within.

Reply:
Previously hitting enter on wrapper navigated to the link but now nothing happens

Reply:
I think, `<a>` should be focusable, but only `focus-within` should be set on the `NcListItem` wrapper. We probably need it anyway, because screen-reader navigation will navigate to the link itself.

Only elements with interactive `role` should have focus.

And if we have entire `NcListItem` focusable, we again have a focusable element inside a focusable element. Not sure if it is fine.

Reply:
> hm, i think we need to make a real focus on the link but visual focus on a whole `NcListItem`, probably like: [nextcloud/server#41439 (comment)](https://github.com/nextcloud/server/pull/41439#issuecomment-1810201036)

Exactly the same solution as in Apps won't work here, because it will overflow the actions button.

But it could be the same as in files, there only file name is an actual link

Reply:
> we again have a focusable element inside a focusable element

That is fine in some cases where you should be able to focus outer wrappers (e.g. in a grid) and then move the focus inside.

But I agree here we probably should not have the wrapper focused.